src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1782,15 +1798,58 @@ RelationRestrictionPartitionKeyIndex(RelationRestriction *relationRestriction)
 			IsPartitionColumn(targetExpression, relationPlannerParseQuery))
 		{
 			Var *targetColumn = (Var *) targetExpression;
-
-			if (targetColumn->varno == relationRestriction->index)
+			/*
+			 * use the rte that we found while traversing the query tree.
+			 */
+			RangeTblEntry* res = findRteIdentityContext->rte;
```

Comment:
shouldn't this be: `			RangeTblEntry* res = list_nth(relationPlannerParseQuery->rtable, targetColumn->varno-1);
`